core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java

```diff
@@ -376,7 +377,8 @@ public Group getWatchGroup() {
       String sequence1 = getCombinedIndexSequences(views.get(i));
       for (int j = i + 1; j < views.size(); j++) {
         String sequence2 = getCombinedIndexSequences(views.get(j));
-        if (Index.checkEditDistance(sequence1, sequence2) < minimumDistance) {
+        if ((Index.checkEditDistance(sequence1, sequence2) < minimumDistance) &&
+            !((hasFakeSequence(views.get(i)) || hasFakeSequence(views.get(j))) && minimumDistance > 1)) {
```

Comment:
The `hasFakeSequence` is a pretty cheap check compared to the `checkEditDistance`, so it would be better to do it first and avoid the more intense check.

Also, a pool shouldn't contain a mix of fake and real sequences. Actually, that might be a new class of error we have to consider. For now, I'd check everything at the start:

`if (views.stream().anyMatch(PoolImpl::hasFakeSequence)) { return Collections.emptySet(); }`